src/cart/swim/swim.c

```diff
@@ -265,6 +266,10 @@ swim_member_alive(struct swim_context *ctx, swim_id_t from,
 		D_GOTO(out, rc);
 	}
 
+	/* Do not widely spread the information about bootstrap complete */
+	if (id_state.sms_status == SWIM_MEMBER_INACTIVE)
+		count = ctx->sc_piggyback_tx_max;
+
```

Comment:
Did the test fail because "rank 7 is dead" messages did not override "rank 7 is alive" messages?

Reply:
Yes, exactly. The message "rank 7 is alive" is from completing bootstrap. So, it spread during the long time and override other information.

Reply:
Should we fix the override rules? There are a discussion on incarnation-based overrides in the SWIM paper, although I don't understand why CONFIRM override any updates with _any_ incarnation.

Reply:
This fix just limit spreading information about first ALIVE event to only single send. Incarnation is not used here because of it will be increased only if self is suspected. But this is a real failure and suspected node is really dead. But in parallel news about it come up from bootstrapping is continue spreading across the cluster. So, this is just fighting between alive from bootstrap and dead immediately.

This patch just reduce the time when the information about bootstrap complete is spreaded. So, the new event with dead immediately after servers bootstrapped will not fight with it.

Reply:
Right, I think I get what this patch intends to do. I'm stuck at: why "rank 7 (incarnation 1) is dead" does not override "rank 7 (incarnation 1) is alive" when they fight?

Reply:
The rank 7 is not dead yet. It should be dead after suspect timeout expired only. But when we start suspect timeout we receive message rank 7 is alive after bootstrap. So, we remove it from suspect list. Then we receive again the message about rank 7 is suspected and start suspect timeout from scratch. Finally we will got suspect timeout expired only when message about rank 7 is alive after bootstrap stop spreading.

Reply:
I see. The protocol requires that "rank 7 (incarnation X) is alive" overrides "rank 7 (incarnation Y) is suspected" only if X > Y. That further requires that the rank who pings rank 7 successfully must carry "rank 7 (incarnation ...) is suspected", and rank 7 must increment its incarnation when replying that ping.

Reply:
This starts to get very interesting. :)

Reply:
Incarnations are the same.

Reply:
Incarnation incremented only if self node detects that somebody suspect it. To avoid wrong guesses it increment incarnation about self only. This is other case when guesses are equivalent and we cannot prioritize one of them.

Reply:
Yes. "rank 7 (incarnation 1) is suspected" can override "rank 7 (incarnation 1) is alive", but the latter can't override the former, according to the protocol.

Reply:
fixed.

Reply:
Thanks for the patience with me. :) The two if statements below have some redundancy that could be simplified, if you need to refresh this patch.